File: core/src/domain/models/resources.rs
use super::resource_groups::ResourceGroupIdentifier;

pub type ResourceIdentifier = String;

#[derive(Debug, Clone, Hash, Eq, PartialEq)]
pub struct ResourceBase {
    pub id: ResourceIdentifier,
    pub display_name: String,
    pub description: Option<String>,
}

#[derive(Debug, Clone, Hash, Eq, PartialEq)]
pub struct RootResource {
    pub base: ResourceBase,
}

#[derive(Debug, Clone, Hash, Eq, PartialEq)]
pub struct ChildResource {
    pub base: ResourceBase,
    pub group_id: ResourceGroupIdentifier,
}

pub enum Resource {
    Root(RootResource),
    Child(ChildResource),
}

File: app/src/application/models/resources.ts
import { ResourceGroupIdentifier } from "./resource_groups";

export type ResourceIdentifier = string;

export type ResourceBase = {
  id: ResourceIdentifier;
  displayName: string;
  description?: string;
};

export type RootResource = ResourceBase & {};

export type ChildResource = ResourceBase & {
  groupId: ResourceGroupIdentifier;
};
